emcc.py

```diff
@@ -447,22 +444,24 @@ def run(args):
   ''' % (shared.EMSCRIPTEN_VERSION, revision))
     return 0
 
-  elif len(args) == 2 and args[1] == '-v': # -v with no inputs
+  shared.check_sanity(force=DEBUG)
```

Comment:
should we be running the sanity check unconditionally, though? this adds overhead on each call to emcc.

Reply:
Its only forced in DEBUG is set.  So I assume that means it doesn't do much?

Anyway, this is a non-functional change, and I'd rather keep it that way.  If we want to change how/when its run I suggest we do that separably.

This change at least makes it a lot more clean when it gets run IMHO.

Reply:
Maybe the "forced" param isn't clear enough - it still checks if it should check the sanity (for example, whether the settings file changed). Force makes it skip that check, but it still does the previous stuff.

Reply:
Yes, but presumably we *do* want to do all that basic stuff on every invocation.    Or at least that is that status quo and I'm not trying to change that with this PR.

Reply:
Oh, I see, we did do this before then. Maybe we can optimize it. Sorry for misreading before.

Reply:
Yes, that is one of the reasons for this refactor.  Its quite hard to tell that the existing code was running check_sanity() in almost all cases, based on the way the condition was expressed.

Reply:
Right, yeah, I see that now. Very nice cleanup.